src/providers/EtherspotContextProvider.tsx

```diff
@@ -276,13 +284,76 @@ const EtherspotContextProvider = ({
     return [];
   }, [sdk, accountAddress]);
 
+  const getSmartWalletBalancesPerChain = useCallback(async (walletAddress: string, supportedChains: Chain[]) => {
+    if (!sdk || !walletAddress) return;
+    let balanceByChain : IBalanceByChain[] = []
+    for (let index = 0; index < supportedChains.length; index++) {
```

Comment:
this would hold each cycle until previous resolved, can't we use `await Promise.all(supportedChains.map(async () => ...`?

Reply:
So the issue with promises.all is that if any one is not resolved then it'll throw error instead of returning other resolved results